Node Maturation Fixes; Docker Documentation Correction; Improved Staking Overview; Ursula JSON status (#1615)
nucypher/nucypher: docs/source/guides/staking_guide.rst

NuCypher Staking Guide
=======================

NuCypher staking operations are divided into two roles:

*Staker* - Controls NU tokens, manages staking, and collects rewards.
The Staker is a manager of one or more stakes and delegates active network participation to a *Worker* through *bonding*.
There is a 1:1 relationship between the roles: A Staker controls a single ethereum account and may have multiple Stakes, but only ever has one Worker.

> Stakers can run on a laptop and do not need to remain online since they only need to send stake management transactions.

Reviewer: ```suggestion
Stakers can run on a laptop and do not need to remain online since they only perform stake management transactions.
```

Hardware wallets are ideal for stakers since they only need to be unlocked during stake management while

> providing a higher standard of security than software wallets.  In order to stake the staker's account

Reviewer: ```suggestion
providing a higher standard of security than software wallets.  In order to stake, the staker's account
```

Author: :heavy_check_mark:

needs both NU token as well as enough ether to pay for transaction gas.

*Worker* - (aka "Ursula") Participates in the network by carrying out re-encryption work orders.
The Worker is the bonded delegate of a Staker and an active network node. Workers must remain online to provide
uninterrupted re-encryption services on-demand. Each staking account or Staker is bonded to exactly one Worker.
The worker's ethereum account must remain unlocked to send automated work confirmation

> transactions and have enough ether to pay for transaction gas; However, It is *not* necessary (and potentially risky)

Reviewer 2: ```suggestion
transactions and have enough ether to pay for transaction gas; however, it is *not* necessary (and potentially risky)
```

Author: :heavy_check_mark:

No action is needed to release the re-staking lock once the release period begins.

Prolong
*******

Existing stakes can be extended by a number of periods as long as the resulting
stake's duration is not longer than the maximum. To prolong an existing stake's duration:

.. code:: bash

    (nucypher)$ nucypher stake prolong

Wind Down
**********

> Wind down is *diabled* by default. To start winding down an existing stake:

Reviewer 2: ```suggestion
Wind down is *disabled* by default. To start winding down an existing stake:
```
Also, it would be good to explain what `wind down` actually means.

Author: :heavy_check_mark:

.. code:: bash

    (nucypher)$ nucypher stake winddown

Divide
******

> Existing stakes can be divided into smaller sub-stakes, with added value and duration. To divide an existing stake:

Reviewer: ```suggestion
Existing stakes can be divided into smaller sub-stakes, with different values and durations. To divide an existing stake:
```

Author: :heavy_check_mark:

.. code:: bash

>     (nucypher)$ nucypher stake divide --hw-wallet

Reviewer 2: Previous commands didn't use the --hw-wallet flag

Author: :heavy_check_mark:
